src/legacy/ui/public/courier/search_strategy/default_search_strategy.js

```diff
@@ -57,68 +58,86 @@ async function serializeAllFetchParams(fetchParams, searchRequests, serializeFet
+  // e.g. body, filters, index pattern, query.
+  const allFetchParams = await getAllFetchParams(searchRequests, Promise);
 
-    const searching = es.msearch(msearchParams);
+  // Serialize the fetch params into a format suitable for the body of an ES query.
+  const {
+    serializedFetchParams,
+    failedSearchRequests,
+  } = await serializeAllFetchParams(allFetchParams, searchRequests, serializeFetchParams);
 
+  if (serializedFetchParams.trim() === '') {
     return {
-      // Munge data into shape expected by consumer.
-      searching: new Promise((resolve, reject) => {
-        // Unwrap the responses object returned by the ES client.
-        searching.then(({ responses }) => {
-          resolve(responses);
-        }).catch(error => {
-          // Format ES client error as a SearchError.
-          const { statusCode, displayName, message, path } = error;
-
-          const searchError = new SearchError({
-            status: statusCode,
-            title: displayName,
-            message,
-            path,
-          });
-
-          reject(searchError);
-        });
-      }),
-      abort: searching.abort,
       failedSearchRequests,
     };
-  },
+  }
+  const msearchParams = {
+    ...getMSearchParams(config),
+    body: serializedFetchParams,
+  };
 
-  isViable: (indexPattern) => {
-    if (!indexPattern) {
-      return false;
-    }
+  const searching = es.msearch(msearchParams);
 
-    return isDefaultTypeIndexPattern(indexPattern);
-  },
-};
+  return {
+    // Munge data into shape expected by consumer.
+    searching: new Promise((resolve, reject) => {
+      // Unwrap the responses object returned by the ES client.
+      searching.then(({ responses }) => {
+        resolve(responses);
+      }).catch(error => {
+        // Format ES client error as a SearchError.
+        const { statusCode, displayName, message, path } = error;
+
+        const searchError = new SearchError({
+          status: statusCode,
+          title: displayName,
+          message,
+          path,
+        });
+
+        reject(searchError);
+      });
+    }),
+    abort: searching.abort,
+    failedSearchRequests,
+  };
+}
+
+function search({ searchRequests, es, Promise, config, sessionId, esShardTimeout }) {
+  const abortController = new AbortController();
+  const searchParams = getSearchParams(config, sessionId, esShardTimeout);
+  const promises = searchRequests.map(async searchRequest => {
+    const { index, body } = searchRequest.fetchParams = await searchRequest.getFetchParams();
+    try {
+      const promise = es.search({ index: index.title || index, body, ...searchParams });
+      abortController.signal.addEventListener('abort', promise.abort);
+      return promise;
+    } catch (e) {
+      return JSON.parse(e.response);
+    }
+  });
+  return {
+    searching: Promise.all(promises),
+    abort: () => abortController.abort(),
+    failedSearchRequests: []
```

Comment:
As I understand `failedSearchRequests` will always be returned empty `[]`. Is it intentional or shall it contain failed requests?